src/api/creatures.py

```diff
@@ -221,44 +229,62 @@ def feed_creature(user_id: int, creature_id: int, treat_sku: str):
                     },
                 )
 
-                connection.execute(
-                    sqlalchemy.text(
-                        """
+            connection.execute(
+                sqlalchemy.text(
+                    """
+                    INSERT INTO user_creature_connection (user_id, creature_id, affinity)
+                    VALUES (:user, :creature, :affinity)
+                    ON CONFLICT(user_id, creature_id)
+                    DO UPDATE 
+                    SET affinity = user_creature_connection.affinity + :affinity
+                """
```

Comment:
There's some inconsistent formatting with the sql queries. This is going to have to be fixed manually.